File: benchmarks/values/int_collatz.py
import time

t0 = time.perf_counter()

total = 0
for n in range(1, 1_000_001):
    x = n
    steps = 0
    while x != 1:
        if x % 2 == 0:
            x = x // 2
        else:
            x = 3 * x + 1
        steps += 1
    total += steps

t1 = time.perf_counter()
print(total)
print(f"elapsed: {t1 - t0:.3f}s")

File: benchmarks/values/int_collatz.rs
use std::time::Instant;

fn main() {
    let t0 = Instant::now();

    let mut sum: i64 = 0;
    for n in 1..=1_000_000i64 {
        let mut x = n;
        let mut steps: i64 = 0;
        while x != 1 {
            if x % 2 == 0 {
                x = x / 2;
            } else {
                x = 3 * x + 1;
            }
            steps += 1;
        }
        sum += steps;
    }

    let elapsed = t0.elapsed();
    println!("{}", sum);
    println!("elapsed: {:.3}s", elapsed.as_secs_f64());
}
